mlflow/ml-package-versions.yml

```diff
@@ -423,7 +423,10 @@ spark:
       # 'pytest tests/spark/autologging') not to share SparkSession across tests
       # `test_spark_datasource_autologging_crossframework.py` fails when run with other tests
       pytest tests/spark/autologging/datasource/test_spark_datasource_autologging_crossframework.py
-      find tests/spark/autologging -name 'test*.py' | grep -v crossframework | xargs -L 1 pytest
+      # `test_spark_disable_autologging.py` is flaky if it runs with other tests
```

Comment:
There are some strange issue in spark session stop / start in the same process that causes test failure here.

Reply:
What are some strange issues?

Reply:
This error:
```
Exception while attempting to initialize JVM-side state for Spark datasource autologging. Note that Spark datasource autologging only works with Spark 3.0 and above. Please create a new Spark session with required Spark version and ensure you have the mlflow-spark JAR attached to your Spark session as described in https://mlflow.org/docs/latest/tracking/autolog.html#spark Exception:
An error occurred while calling z:org.mlflow.spark.autologging.MlflowAutologEventPublisher.init.
: java.lang.RuntimeException: Unable to get active SparkSession. Please ensure you've started a SparkSession via SparkSession.builder.getOrCreate() before attempting to initialize Spark datasource autologging.
	at org.mlflow.spark.autologging.MlflowAutologEventPublisherImpl.$anonfun$spark$1(MlflowAutologEventPublisher.scala:42)
	at scala.Option.getOrElse(Option.scala:189)
	at org.mlflow.spark.autologging.MlflowAutologEventPublisherImpl.spark(MlflowAutologEventPublisher.scala:40)
	at org.mlflow.spark.autologging.MlflowAutologEventPublisherImpl.spark$(MlflowAutologEventPublisher.scala:39)
	at org.mlflow.spark.autologging.MlflowAutologEventPublisher$.spark(MlflowAutologEventPublisher.scala:20)
	at org.mlflow.spark.autologging.MlflowAutologEventPublisherImpl.isInReplAwareContext(MlflowAutologEventPublisher.scala:52)
	at org.mlflow.spark.autologging.MlflowAutologEventPublisherImpl.getSparkDataSourceListener(MlflowAutologEventPublisher.scala:71)
	at org.mlflow.spark.autologging.MlflowAutologEventPublisherImpl.getSparkDataSourceListener$(MlflowAutologEventPublisher.scala:70)
	at org.mlflow.spark.autologging.MlflowAutologEventPublisher$.getSparkDataSourceListener(MlflowAutologEventPublisher.scala:20)
	at org.mlflow.spark.autologging.MlflowAutologEventPublisherImpl.init(MlflowAutologEventPublisher.scala:82)
	at org.mlflow.spark.autologging.MlflowAutologEventPublisherImpl.init$(MlflowAutologEventPublisher.scala:80)
	at org.mlflow.spark.autologging.MlflowAutologEventPublisher$.init(MlflowAutologEventPublisher.scala:20)
	at org.mlflow.spark.autologging.MlflowAutologEventPublisher.init(MlflowAutologEventPublisher.scala)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:244)
	at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:374)
	at py4j.Gateway.invoke(Gateway.java:282)
	at py4j.commands.AbstractCommand.invokeMethod(AbstractCommand.java:132)
	at py4j.commands.CallCommand.execute(CallCommand.java:79)
	at py4j.GatewayConnection.run(GatewayConnection.java:238)
	at java.base/java.lang.Thread.run(Thread.java:840)
```
it is raised from this line:
https://github.com/mlflow/mlflow/blob/a92e8164b3cf88fc38bbb3f85b559fc943d912f1/mlflow/java/spark/src/main/scala/org/mlflow/spark/autologging/MlflowAutologEventPublisher.scala#L42

It is probably race condition issue inside `sparkSession` implementation, and it might be triggered if we shut down spark session and then start a new session in the same process.